gc/base/MemoryPoolAddressOrderedList.cpp

```diff
@@ -634,9 +686,31 @@ MM_MemoryPoolAddressOrderedList::internalAllocateTLH(MM_EnvironmentBase *env, ui
 	consumedSize = (maximumSizeInBytesRequired > freeEntrySize) ? freeEntrySize : maximumSizeInBytesRequired;
```

Comment:
consider having these 3 lines along with the new block down below that also can mutate consumedSize going together into a new method so that we simplify the code here:
if (0 == (consumedSize = determineConsumedSizeForTLH(&freeEntry)) 
retry;

decrementFreeEntrySizeClassStats should go either before or after that code, not inside

but remember that as per one of earlier comments, I suspect that there is a missing logic where even freeEntry pointer can be changed.....  so determineConsumedSizeForTLH should evolve to cover that case, too

Reply:
>I suspect that there is a missing logic where even freeEntry pointer can be changed..... so determineConsumedSizeForTLH should evolve to cover that case, too

I'm not sure about this, could you please clarify

Reply:
Ah yes, I didn't addresses that yet so wasn't considering this case. But that makes sense, essentially clipping the tlh from the other end

Reply:
I'm willing to address that case a followup change (but would wise to add a todo as a comment and/or raise an issue).